File: lista-de-tarefas/src/models/Tarefa.ts
import { Categoria } from "./Categoria";
import { SubTarefa } from "./SubTarefa";

export interface Tarefa {
  tarefaId: string;
  titulo: string;
  descricao: string;
  criadoEm: string;
  categoriaId?: string;
  categoria?: Categoria;
  status: string;
  usuarioId: string;
  subTarefas?: SubTarefa[];
}

File: TarefasApi/Models/Tarefa.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace TarefasApi.Models;

public class Tarefa
{
    [Key]
    public string TarefaId { get; set; } = Guid.NewGuid().ToString();
    public string Titulo { get; set; }
    public string Descricao { get; set; }
    public DateTime CriadoEm { get; set; } = DateTime.Now;
    public string UsuarioId { get; set; }
    public Usuario Usuario { get; set; }
    public Categoria? Categoria { get; set; }
    public string? CategoriaId { get; set; }
    public string? Status { get; set; } = "Não iniciada";
}
